### PR TITLE
[master] Add version param to pkg_params in _uninstall()

### LINKED ISSUE #658: [BUG] Zypperpkg does not use the version parameter
**Description**
When using the `pkg.removed` function with the `version` parameter to remove a specific package version, the `zypperpkg` module does not use the `version` parameter to remove only that specified package. The assumption is that the pkg module verifies that the version is installed in [0], and the zypperpkg module then removes the package without using a version in [1].

This assumption does not hold for RPM packages that enable multiple versions installed at the same time, such as the kernel package.  In such cases, zypperpkg removes the package by using its name, which results in both kernel versions being removed.

For example, this SLS:

```
remove_previous_kernel_default:
  pkg.removed:
    - name: kernel-default
    - version: 6.4.0-150600.21.1
```

results in no Kernels installed:

```
localhost:~ # rpm -qa | grep kernel-default
kernel-default-6.9.1-1.1.x86_64
kernel-default-6.4.0-150600.21.1.x86_64
localhost:~ # salt-call --local state.sls test
local:
----------
          ID: remove_previous_kernel-default
    Function: pkg.removed
        Name: kernel-default
      Result: True
     Comment: All targeted packages were removed.
     Started: 17:03:13.430864
    Duration: 3788.295 ms
     Changes:   
              ----------
              kernel-default:
                  ----------
                  new:
                  old:
                      6.4.0-150600.21.1,6.9.1-1.1
localhost:~ # rpm -qa | grep kernel-default
```

[0] https://github.com/openSUSE/salt/blob/openSUSE/release/3006.0/salt/states/pkg.py#L2903-L2905 
[1] https://github.com/openSUSE/salt/blob/openSUSE/release/3006.0/salt/modules/zypperpkg.py#L2150

**Expected behavior**
Only the specified RPM package version gets removed.

### PR DESCRIPTION
**Pull request by winddss**
### What does this PR do?

Add "version" param to pkg_params from parse_targes() in _uninstall(), to check the pkg list after the specific version of the package removed.

### What issues does this PR fix or reference?
Fixes: https://github.com/openSUSE/salt/issues/658
Reference: https://github.com/saltstack/salt/commit/585145f4b382508ef7a6cc6f1cb33697b9a606b8
Reference: https://github.com/saltstack/salt/pull/66713/commits/c0d28890c6163861bdcb73d2cec7084e262c4e5b

Need for PR: https://github.com/saltstack/salt/pull/66713

### Previous Behavior
pkg_params only have the package name

### New Behavior
pkg_params will include the package name:version

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by Akm0d — APPROVED** [review]

**Comment by winddss** [comment]
@Akm0d the failed in "CI / Arch Linux LTS Test / Test (zeromq, functional) (pull_request)" just caused by a GPG error when installing the package, it seems not relate to my patch, could you please re-run the test? thanks.

**Comment by winddss** [comment]
@Akm0d the TEST still failed with "CI / Arch Linux LTS Test / Test (zeromq, functional) (pull_request)", the error followng:

> Problem encountered installing package(s). Additional info follows:\n\nerrors:\n    - Running as unit: run->ra20679281f924e8fa27aea22ab049893.scope; invocation ID: a4846cdc25f849a0ae6dfdf09a4c2ce0\n      error: sl: >signature from "Brett Cornwall <brett@i--b.com>" is unknown trust\n      error: failed to commit transaction (invalid or >corrupted package (PGP signature))', '__sls__': None, '__run_num__': 0, 'start_time': '20:34:42.654470', 'duration': 1265.074, >'__id__': None}}, state_id='pkg_|-None_|-None_|-installed', full_return={'name': None, 'result': False, 'changes': {}, >'comment': 'Problem encountered installing package(s). Additional info follows:\n\nerrors:\n  

It seems someone's PGP signature expired to caused the package install failed.

File: salt/states/pkg.py [hunk 1]
```diff
@@ -2964,7 +2964,7 @@ def _uninstall(
 
     try:
         pkg_params = __salt__["pkg_resource.parse_targets"](
-            name, pkgs, normalize=normalize
+            name, pkgs, normalize=normalize, version=version
```
Comment (twangboy) [inline salt/states/pkg.py:2967]: Looks like this is fixed here: https://github.com/saltstack/salt/blob/3006.x/salt/states/pkg.py#L2887

**Comment by twangboy** [comment]
Already added to 3006.x branch: https://github.com/saltstack/salt/pull/66810

**Pull request closed without merging**